mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/blob/BlobStoreDeletedMessageVault.java

```diff
@@ -119,7 +120,9 @@
         return Flux.from(messageMetadataVault.listMessages(bucketName, user))
             .filter(deletingInformation -> messageId.equals(deletingInformation.getDeletedMessage().getMessageId()))
             .flatMap(deletingInformation -> deleteMessageData(bucketName, deletingInformation)
-                .then(deleteMessageMetaData(bucketName, user, messageId)))
+                .then(deleteMessageMetaData(bucketName, user, messageId))
+                .thenReturn(deletingInformation))
+            .flatMap(this::deleteSameBlobInOtherBuckets)
```

Comment:
This step is very surprising (I'm not saying that it's not what we decided but that a developer calling this function won't figure out about this behavior). Don't you think it requires to be moved at a higher-level ? I mean, it's clearly some business logic

Reply:
If I move it to outside, do it in another component, but I still need to know:
 - list of buckets in the vault
 - the blobId associated with the deleted message
 - use the blobStore to delete the blob by (bucketName + blobId)
So, the component does this, it requires to have knowledge about blobStore & vault and exposes the abstraction of the relationship between vault + blobStore we want to hide. 

I still want to do in in the vault, but in a new API method like `cleanData(User, MessageId)` and keep the delete() as it is, WDYT ?

Reply:
What is the functional description of this last step?

Reply:
the final conclusion is remote that last step!